# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/src/picture_graph.rs
# Commit: aaa7e1f81404
# Full Hash: aaa7e1f8140455a9f83be8278720598ea3fe4cd8
# Author: Glenn Watson <git@intuitionlibrary.com>
# Date: 2021-12-14 09:42:05
# Description:
#   Bug 1745775 - Fix invalid surface info indexing r=gfx-reviewers,bradwerth
#   
#   Some parts of the visibility pass have been ported to use the
#   picture graph infrastructure (update pass assignment and bounding
#   rect propagation) but the main visibility pass still relies on
# ==============================================================================

diff -r de1608e5f69b -r aaa7e1f81404 gfx/wr/webrender/src/picture_graph.rs
--- a/gfx/wr/webrender/src/picture_graph.rs	Tue Dec 14 00:24:22 2021 +0000
+++ b/gfx/wr/webrender/src/picture_graph.rs	Tue Dec 14 01:06:24 2021 +0000
@@ -162,6 +162,9 @@
 
     info.parent = parent_pic_index;
 
+    // Run pre-update to resolve animation properties etc
+    pic.pre_update(frame_context);
+
     let can_be_drawn = match info.update_pass {
         Some(update_pass) => {
             // No point in recursing into paths in the graph if this picture already
@@ -174,7 +177,7 @@
         }
         None => {
             // Check if this picture can be dropped from the graph we're building this frame
-            pic.pre_update_visibility_check(frame_context)
+            pic.is_visible(frame_context.spatial_tree)
         }
     };
 